components/ntp_background_images/browser/ntp_sponsored_images_data.cc

```diff
@@ -196,83 +135,230 @@ void NTPSponsoredImagesData::ParseCampaignsList(
-    default_logo = GetLogoFromValue(installed_dir, url_prefix, *logo);
+  const std::string* const campaign_id = value.FindString(kCampaignIdKey);
+  if (!campaign_id) {
+    // Campaign ID is required.
+    return std::nullopt;
   }
+  campaign.campaign_id = *campaign_id;
 
-  if (auto* wallpapers = value.FindList(kWallpapersKey)) {
-    for (const auto& entry : *wallpapers) {
-      const auto& wallpaper = entry.GetDict();
-      const std::string* image_url = wallpaper.FindString(kImageURLKey);
-      if (!image_url) {
-        continue;
-      }
+  const base::Value::List* const creative_sets =
+      value.FindList(kCreativeSetsKey);
+  if (!creative_sets) {
+    // Creative sets are required.
+    return std::nullopt;
+  }
+
+  for (const auto& creative_set : *creative_sets) {
+    const base::Value::Dict* const creative_set_dict = creative_set.GetIfDict();
+    if (!creative_set_dict) {
+      // Invalid creative set.
+      return std::nullopt;
+    }
+
+    const std::string* const creative_set_id =
+        creative_set_dict->FindString(kCreativeSetIdKey);
+    if (!creative_set_id) {
+      // Creative set ID is required.
+      return std::nullopt;
+    }
+
+    const base::Value::List* const creatives =
+        creative_set_dict->FindList(kCreativesKey);
+    if (!creatives) {
+      // Creative are required.
+      return std::nullopt;
+    }
 
+    for (const auto& creative : *creatives) {
       SponsoredBackground background;
-      background.file_path = installed_dir.AppendASCII(*image_url);
 
-      if (auto* focal_point = wallpaper.FindDict(kWallpaperFocalPointKey)) {
-        background.focal_point = {focal_point->FindInt(kXKey).value_or(0),
-                                  focal_point->FindInt(kYKey).value_or(0)};
+      const base::Value::Dict* const creative_dict = creative.GetIfDict();
+      if (!creative_dict) {
+        // Invalid creative.
+        return std::nullopt;
       }
 
-      if (const auto* const condition_matchers =
-              wallpaper.FindList(kWallpaperConditionMatchersKey)) {
+      const std::string* const creative_instance_id =
+          creative_dict->FindString(kCreativeInstanceIdKey);
+      if (!creative_instance_id) {
+        // Creative instance ID is required.
+        return std::nullopt;
+      }
+      background.creative_instance_id = *creative_instance_id;
+
+      const std::string* const company_name =
+          creative_dict->FindString(kCreativeCompanyNameKey);
+      if (!company_name) {
+        // Company name is required.
+        return std::nullopt;
+      }
+      background.logo.company_name = *company_name;
+
+      const std::string* const alt = creative_dict->FindString(kCreativeAltKey);
+      if (!alt) {
+        // Alt is required.
+        return std::nullopt;
+      }
+      background.logo.alt_text = *alt;
+
+      const std::string* const target_url =
+          creative_dict->FindString(kCreativeTargetUrlKey);
+      if (!target_url) {
```

Comment:
may be a good place to validate target_url protocol is HTTPS

Reply:
Actually for this PR I will just validate against https, and will followup with support as above. Thanks

Reply:
Fixed